seq_logger.js

```diff
@@ -249,76 +268,85 @@ class SeqLogger {
         this._queue.splice(0, i);
         return {batch, bytes};
     }
-    
+
+    _httpOrNetworkError(res) {
+        const networkErrors = ['ECONNRESET', 'ENOTFOUND', 'ESOCKETTIMEDOUT', 'ETIMEDOUT', 'ECONNREFUSED', 'EHOSTUNREACH', 'EPIPE', 'EAI_AGAIN', 'EBUSY'];
+        return networkErrors.includes(res) || 500 <= res.statusCode && res.statusCode < 600;
+    }
+
     _post(batch, bytes) {
-        return new Promise((resolve, reject) => {
-            var opts = {
-                host: this._endpoint.hostname,
-                port: this._endpoint.port,
-                path: this._endpoint.path,
-                protocol: this._endpoint.protocol,
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Content-Length': bytes
-                },
-                timeout: this._requestTimeout
-            };
-            
-            if (this._apiKey) {
-                opts.headers["X-Seq-ApiKey"] = this._apiKey;
-            }
+        let attempts = 0;
 
-            let requestFactory = opts.protocol === 'https:' ? https : http;
-            let req = requestFactory.request(opts);
+        return new Promise((resolve, reject) => {
+            const sendRequest = (batch, bytes) => {
+                attempts++;
+                let requestFactory = this._endpoint.protocol === "https:" ? https : http;
+                let req = requestFactory.request({
```

Comment:
Ok, I had a better look at this, and we can use the empty `http.Agent` shim, but will just need to include the same request configuration (`host`, `port`, `protocol`, `headers`) here as well as on the agent so that we don't lose browser support.

Sorry this is a bit unclear, we should look at including a sample browser app and run it in CI sometime.